nemo/collections/nlp/data/text_normalization/decoder_dataset.py

```diff
@@ -0,0 +1,151 @@
+        s_input = s_left + [extra_id_0] + c_s_words + [extra_id_1] + s_right
+        if inst_dir == constants.INST_BACKWARD:
+            input_words = [constants.ITN_PREFIX] + s_input
+            output_words = c_w_words
+        if inst_dir == constants.INST_FORWARD:
+            input_words = [constants.TN_PREFIX] + w_input
+            output_words = c_s_words
+        # Finalize
+        self.input_str = ' '.join(input_words)
+        self.output_str = ' '.join(output_words)
+        self.direction = inst_dir
+        self.semiotic_class = semiotic_class
+
+class TextNormalizationDecoderDataset:
+    def __init__(
+            self,
+            input_file: str,
+            tokenizer: PreTrainedTokenizerBase,
+            mode: str,
+            max_len: int,
+            decoder_data_augmentation: bool
+        ):
+        assert(mode in constants.MODES)
+        self.mode = mode
+        raw_insts = read_data_file(input_file)
+
+        # Convert raw instances to TaggerDataInstance
+        insts, inputs, targets = [], [], []
+        for (classes, w_words, s_words) in tqdm(raw_insts):
+            for ix, (_class, w_word, s_word) in enumerate(zip(classes, w_words, s_words)):
+                if s_word in constants.SPECIAL_WORDS: continue
+                for inst_dir in constants.INST_DIRECTIONS:
+                    if inst_dir == constants.INST_BACKWARD and mode == constants.TN_MODE: continue
+                    if inst_dir == constants.INST_FORWARD and mode == constants.ITN_MODE: continue
+                    # Create a DecoderDataInstance
+                    inst = DecoderDataInstance(w_words, s_words, inst_dir,
+                                               start_idx=ix, end_idx=ix+1,
+                                               semiotic_class=_class)
+                    insts.append(inst)
+                    if decoder_data_augmentation:
+                        noise_left = random.randint(1, 2)
+                        noise_right = random.randint(1, 2)
+                        inst = DecoderDataInstance(w_words, s_words, inst_dir,
+                                                   start_idx=ix-noise_left,
+                                                   end_idx=ix+1+noise_right)
+                        insts.append(inst)
+
+        self.insts = insts
+        inputs = [inst.input_str for inst in insts]
+        targets = [inst.output_str for inst in insts]
+
+        # Tokenization
+        self.inputs, self.examples = [], []
+        self.tn_count, self.itn_count, long_examples_filtered = 0, 0, 0
+        input_max_len, target_max_len = 0, 0
+        for idx in range(len(inputs)):
+            # Input
+            _input = tokenizer([inputs[idx]])
+            input_len = len(_input['input_ids'][0])
+            if input_len > max_len:
+                long_examples_filtered += 1
+                continue
+
+            # Target
+            with tokenizer.as_target_tokenizer():
```

Comment:
what does this do?

Reply:
This function is from HuggingFace. I just checked and it did not make any difference so I removed it.